decred/examples/drain_xpriv.py

```diff
@@ -0,0 +1,255 @@
+    """
+    Create one trasaction spending all the outputs to the passed address and sign the
+    inputs.
+    """
+    payToScript = txscript.payToAddrScript(sendToAddr)
+    output = msgtx.TxOut(value=0, version=0, pkScript=payToScript)
+    inputs = []
+    for utxo in utxos:
+        opCodeClass = txscript.getP2PKHOpCode(utxo["script"])
+        tree = (
+            wire.TxTreeRegular
+            if opCodeClass == txscript.opNonstake
+            else wire.TxTreeStake
+        )
+        op = msgtx.OutPoint(txHash=utxo["hash"], idx=utxo["n"], tree=tree)
+        txIn = msgtx.TxIn(previousOutPoint=op, valueIn=int(utxo["value"] * 1e8))
+        inputs.append(txIn)
+
+    newTx = msgtx.MsgTx(
+        serType=wire.TxSerializeFull,
+        version=txscript.generatedTxVersion,
+        txIn=inputs,
+        txOut=[output],
+        lockTime=0,
+        expiry=0,
+        cachedHash=None,
+    )
+    fee = txscript.calcMinRequiredTxRelayFee(FEE_RATE, newTx.serializeSize())
+    if fee > totalValue:
+        raise DecredError("Not enough funds to cover the transaction fee.")
+
+    newTx.txOut[0].value = totalValue - int(fee)
+
+    for idx, utxo in enumerate(utxos):
+        signatureScript, _, _, _ = txscript.sign(
+            net,
+            newTx,
+            idx,
+            utxo["script"],
+            txscript.SigHashAll,
+            account.KeySource(priv=lambda _: utxo["privKey"], internal=None),
+            crypto.STEcdsaSecp256k1,
+        )
+        newTx.txIn[idx].signatureScript = signatureScript
+
+    return newTx
+
+
+def main():
+    net = None
+    isTestnet = False
+    tString = ""
+    netStr = input("Is this mainnet or testnet?\n")
+    if netStr in ["testnet", "test", "t"]:
+        net = nets.testnet
+        isTestnet = True
+        tString = "t"
+    elif netStr in ["mainnet", "main", "m"]:
+        net = nets.mainnet
+    else:
+        raise DecredError("Unknown network entered.")
+
+    xprivStr = input("Enter xpriv: ")
+    xpriv = decodeExtendedKey(net, xprivStr)
+    # Double check that we can reproduce the xpriv.
+    if b58encode(xpriv.serialize().b).decode() != xprivStr:
+        raise DecredError("unknown xpriv parsing error")
+
+    internal = xpriv.child(INTERNAL)
+    external = xpriv.child(EXTERNAL)
+
+    dcrdConfig = cfg(isTestnet)
+    node = rpc.Client(
+        urlunsplit(("https", dcrdConfig["rpclisten"], "/", "", "")),
+        dcrdConfig["rpcuser"],
+        dcrdConfig["rpcpass"],
+        dcrdConfig["rpccert"],
+    )
+
+    utxos = getUTXOs(node, internal, net) + getUTXOs(node, external, net)
+    totalValue = sum((utxo["value"] for utxo in utxos))
+
+    if totalValue == 0:
+        print("No funds found to send.")
+        return
+
+    print(f"Found {len(utxos)} outputs totalling {totalValue} {tString}dcr.\n")
+
+    sendToAddrStr = input("Input address to send funds: ")
+    # Will throw if bad addr.
+    sendToAddr = addrlib.decodeAddress(sendToAddrStr, net)
```

Comment:
Consider doing this in a loop and catching any exceptions until you get a valid address. This will prevent re-running the entire scan if they flub their address somehow.